src/rpc/net.cpp

```diff
@@ -322,6 +323,61 @@ static RPCHelpMan addnode()
     };
 }
 
+static RPCHelpMan addconnection()
+{
+    return RPCHelpMan{"addconnection",
+        "\nOpen an outbound connection to a specified node. This RPC is for testing only.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
+            {"connectiontype", RPCArg::Type::STR, RPCArg::Optional::NO, "'outbound' or 'blockrelay'."},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                { RPCResult::Type::STR, "address", "Address of newly added connection." },
+                { RPCResult::Type::STR, "connectiontype", "Type of connection opened." },
+            }},
+        RPCExamples{
+            HelpExampleCli("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+            + HelpExampleRpc("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    if (!Params().IsMockableChain()) {
+        throw std::runtime_error("addconnection is for regression testing (-regtest mode) only.");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VSTR});
+    const std::string address = request.params[0].get_str();
+    const std::string conn_type_in = request.params[1].get_str();
+    ConnectionType conn_type{};
+    if (conn_type_in == "outbound") {
+        conn_type = ConnectionType::OUTBOUND_FULL_RELAY;
+    } else if (conn_type_in == "blockrelay") {
+        conn_type = ConnectionType::BLOCK_RELAY;
+    } else {
+        throw std::runtime_error(self.ToString());
+    }
+
+    NodeContext& context = EnsureNodeContext(request.context);
+    if (!context.connman) {
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled.");
+    }
+
+    const bool success = context.connman->AddConnection(address, conn_type);
+    if (!success) {
+        throw JSONRPCError(RPC_CLIENT_NODE_CAPACITY_REACHED, "Error: Already at capacity for specified connection type.");
+    }
+
+    UniValue info(UniValue::VOBJ);
+    info.pushKV("address", address);
+    info.pushKV("connectiontype", conn_type_in);
+
```

Comment:
0c4d673716f2eb37efdb81ad2d7d20f69de960b3 wishlist :)
```diff
+    info.pushKV("peer id", number);
+    info.pushKV("result", "...string description...");
```

Reply:
what would be described in result?

Reply:
also, what are some use cases you have in mind for returning the peer's id?

Reply:
Off-hand, I imagined the peer id might be useful for testing, but that can wait until there is an actual use case or need for it. The result might be a success (`"result": "connection successful"`) or failure (`"error": "unable to connect"`), but I'm not sure how much `addconnection` would be used manually in the CLI or like https://github.com/bitcoin/bitcoin/pull/19315#discussion_r528979140 that returns the object (I didn't look at how to introspect  test_framework.p2p.P2PInterface objects yet). At any rate, I don't want to hold this up. We need more p2p testing tools.

Reply:
okay, I'm going to resolve this comment. we can revisit once there are specific use cases